### PR TITLE
emscripten_3_1_49: init at 3.1.49

### PR DESCRIPTION
**Pull request by willcohen**
## Description of changes

Adds a pinned version of emscripten at 3.1.49 to allow GHC to continue to build.

@sternenseemann I have no interaction with that ecosystem, but if you can let me know which packages I should change the dependency on, I can add them to this PR for ofborg to check!

@RaitoBezarius 

Per https://github.com/NixOS/nixpkgs/pull/269904#issuecomment-1873937282

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/compilers/emscripten/3.1.49.nix [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    # when building the javascript backend.
+    targetPrefix = "em";
+    bintools = emscripten;
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/emscripten-core/emscripten";
+    description = "An LLVM-to-JavaScript Compiler";
+    platforms = platforms.all;
+    maintainers = with maintainers; [ qknight matthewbauer raitobezarius willcohen ];
```
Comment (RaitoBezarius) [inline pkgs/development/compilers/emscripten/3.1.49.nix:138]: I recommend keeping just us two, @sternenseemann would you co-maintain with us this version?

**Comment by RaitoBezarius** [comment]
Looks sane to me, @sternenseemann let us know if this is the way you prefer us to proceed or not.

**Comment by willcohen** [comment]
Updated the maintainers to just be the two of us. @sternenseemann, happy to add you as well if you'd like!

**Comment by voidus** [comment]
It seems like this is not enough to fix the GHC build, so I created the linked issue.

Should the passthrough in the derivation also refer to `emscripten_3_1_49` by the way? If not, the current emscripten is pulled in and stuff fails at another point as well.

**Comment by sternenseemann** [comment]
Obsoleted by #302887.

**Pull request closed without merging**